knative-operator/pkg/webhook/knativeserving/webhook_mutating.go

```diff
@@ -37,14 +33,30 @@ func (v *Configurator) Handle(ctx context.Context, req admission.Request) admiss
 		return admission.Errored(http.StatusBadRequest, err)
 	}
 
-	err = common.Mutate(ks, v.client)
-	if err != nil {
-		return admission.Errored(http.StatusInternalServerError, err)
-	}
+	// Unset the entire registry section. We used to override it anyway, so there can't
+	// be any userdata in there.
+	// TODO: Remove in the 1.21 release to potentially make this usable for users.
+	ks.Spec.CommonSpec.Registry.Override = nil
+
+	defaultToKourier(ks)
 
 	marshaled, err := json.Marshal(ks)
 	if err != nil {
 		return admission.Errored(http.StatusInternalServerError, err)
 	}
 	return admission.PatchResponseFromRaw(req.AdmissionRequest.Object.Raw, marshaled)
 }
+
+// Technically this does nothing in terms of behavior (the default is assumed in the
+// extension code of openshift-knative-operator already), but it fixes a UX nit where
+// Kourier would be shown as enabled: false to the user if the ingress object is
+// specified.
+func defaultToKourier(ks *operatorv1alpha1.KnativeServing) {
```

Comment:
why not in the "extension" ?

Reply:
You read the comment? This is supposed to be user-visible so it has to be in the webhook.

Reply:
yes, but I think your latest comment here did make clear to me, why this is (still) needed